client/src/app/components/manage-organizations/organization-view/organization-view.component.ts

```diff
@@ -36,7 +41,58 @@ export abstract class OrganizationViewComponent implements OnInit {
     }
   }
 
+  populateExcelElem(): void {
+    for (const elem of this.linesR) {
+      this.locations.push(elem);
+    }
+  }
+
+  populateUserInputElem(): void {
+    const locArr = this.location.split(',');
+    // tslint:disable-next-line:prefer-for-of
+    for (let i = 0; i < locArr.length; i += 2) {
+      if (i !== locArr.length - 1) {
+        const tempArr = [];
+        tempArr.push(locArr[i]);
+        tempArr.push(locArr[i + 1]);
+        // @ts-ignore
+        this.locations.push(tempArr);
+      }
+    }
+  }
+
+  // tslint:disable-next-line:typedef
+  public changeListener(files: FileList) {
+    // @ts-ignore
+    // tslint:disable-next-line:no-non-null-assertion
+    this.locationFileName = files.item(0).name!;
+    this.linesR = [];
+    const reader = new FileReader();
+    // @ts-ignore
+    reader.readAsArrayBuffer(files.item(0));
+    reader.onload = () => {
+      const arrayBuffer = reader.result;
+      // @ts-ignore
+      const data = new Uint8Array(arrayBuffer);
+      const arr = [];
+      for (let i = 0; i !== data.length; ++i) {
+        arr[i] = String.fromCharCode(data[i]);
+      }
+      const workbook = XLSX.read(arr.join(''), {type: 'binary'});
+      const worksheet = workbook.Sheets[workbook.SheetNames[0]];
+      const jsonFormat = XLSX.utils.sheet_to_json(worksheet, {raw: true, blankrows: false});
+      // tslint:disable-next-line:prefer-for-of
+      for (let i = 0; i < jsonFormat.length; i++) {
+        const tarrR = [];
+        // @ts-ignore
+        for (const [key, value] of Object.entries(jsonFormat[i])) {
+          tarrR.push(value);
+        }
+        this.linesR.push(tarrR);
+      }
+    };
+  }
+
```

Comment:
Could you be more specific here with the method name ? Or add comments as to what exactly this is doing.

Reply:
So as soon as the excel is uploaded it will be loaded in the array, before submitting.

Reply:
I have renamed this method to readFromExcelFile